src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -28,49 +28,60 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Order;
 
+use Cache;
 use Cart;
+use CartRule;
+use Context;
 use Currency;
 use Order;
+use OrderCartRule;
+use OrderInvoice;
+use PrestaShop\PrestaShop\Core\Cart\CartRuleData;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
 use PrestaShop\PrestaShop\Core\Localization\CLDR\ComputingPrecision;
+use PrestaShopDatabaseException;
+use PrestaShopException;
 use Tools;
 
 class OrderAmountUpdater
 {
     /**
      * @param Order $order
      * @param Cart $cart
-     * @param bool $hasInvoice
+     * @param int|null $orderInvoiceId
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
      */
     public function update(
         Order $order,
         Cart $cart,
-        bool $hasInvoice
-    ): Order {
+        ?int $orderInvoiceId
+    ): void {
```

Comment:
@jolelievre  it is a BC ... right ?

Reply:
It would indeed have been a BC if the class `OrderAmountUpdater` already existed in previous prestashop versions, but this class was introduced in 1.7.7, which is not released yet, so we're fine 👍

Reply:
Yep, it's the last occasion we have to make it right ^^